pixeltable/catalog/catalog.py

```diff
@@ -232,22 +234,30 @@ def validate(self) -> None:
                 f'snapshot_id={tbl_version.id} mutable_views={tbl_version.mutable_views}'
             )
 
-            if tbl_version.is_view and tbl_version.is_mutable and tbl_version.is_validated:
+            assert anchor_tbl_id is None or tbl_version.is_replica
```

Comment:
This should be: is_replica == (anchor_tbl_id is not None)

Reply:
When is that the case? Can we avoid that? Having clearly defined semantics makes avoiding bugs much easier.

Reply:
For example, if a _system table that only has fragments becomes orphaned because its visible descendant was dropped. We need to be able to construct a TableVersion reference to that system table so that we can garbage collect it, but there is no longer anything it can anchor to.

The semantics are perfectly clear; if anchor_tbl_id=None then the TableVersion points to HEAD as always. Such a TableVersion will never result from the user doing a `get_table()`, but that doesn't imply there's any ambiguity,